src/lazy/binary/raw/v1_1/value.rs

```diff
@@ -222,7 +222,33 @@ impl<'top> LazyRawBinaryValue_1_1<'top> {
 
     /// Helper method called by [`Self::read`]. Reads the current value as a float.
     fn read_float(&self) -> ValueParseResult<'top, BinaryEncoding_1_1> {
-        todo!();
+        debug_assert!(self.encoded_value.ion_type() == IonType::Float);
+
+        let value = match self.encoded_value.value_length {
+            8 => {
+                let mut buffer = [0; 8];
+                let val_bytes = self.input.slice(1, 8);
+                buffer[..8].copy_from_slice(val_bytes.bytes());
+
+                f64::from_le_bytes(buffer)
+            }
+            4 => {
+                let mut buffer = [0; 4];
+                let val_bytes = self.input.slice(1, 4);
+                buffer[..4].copy_from_slice(val_bytes.bytes());
+
+                f32::from_le_bytes(buffer).into()
+            }
+            2 => todo!("implement half-precision floats"),
```

Comment:
Can we create an issue to track this? It would be great for users (and our own benchmarks) if we could use 2-byte floats for "common" numbers like `1.0`.

Reply:
Created: #764